pilot/cmd/pilot-agent/main.go

```diff
@@ -439,8 +446,6 @@ func init() {
 		"Proxy unique ID. If not provided uses ${POD_NAME}.${POD_NAMESPACE} from environment variables")
 	proxyCmd.PersistentFlags().StringVar(&role.DNSDomain, "domain", "",
 		"DNS domain suffix. If not provided uses ${POD_NAMESPACE}.svc.cluster.local")
-	proxyCmd.PersistentFlags().StringVar(&trustDomain, "trust-domain", "",
```

Comment:
Is it safe ? Why is it combined with the cred fetcher ?

Reply:
It's already set by the environment variable.

Reply:
But is the injection setting it ? How about VMs systemd command ? We have documented the flags - can't just removed them in unrelated changes and without some discussion. 

Last discussion the agreement was to have small focused PRs. I have no problem with removing the trust-domain flag - and standardizing on ProxyConfig, but not mixed with a controversial interface GCP VMs.

Reply:
I removed it based on the previous comments. If you want to keep it, I don't mind leaving it here. The goal is this PR is not to remove trust domain flag, but to add credential fetcher to support non-k8s workloads.

Reply:
I changed it back. We can do it in a separate change.